microsoftteams_connector.py

```diff
@@ -485,29 +494,53 @@ def _update_request(self, action_result, endpoint, headers=None, params=None, da
                         'Accept': 'application/json',
                         'Content-Type': 'application/json'})
 
-        ret_val, resp_json = self._make_rest_call(action_result=action_result, endpoint=endpoint, headers=headers,
-                                                  params=params, data=data, method=method)
+        status, resp_json = self._make_rest_call(
+            action_result=action_result,
+            endpoint=endpoint,
+            headers=headers,
+            params=params,
+            data=data,
+            method=method
+        )
 
         message = action_result.get_message().lower()
 
-        # If token is expired, generate new token
-        if "token" in message and "expired" in message:
-            status = self._generate_new_access_token(action_result=action_result, data=token_data)
-
-            if phantom.is_fail(status):
+        if phantom.is_fail(status):
+            # If token is expired, generate new token
+            if "token" in message and "expired" in message:
```

Comment:
How does the message currently look like? Because this is quite a simple heuristic

Reply:
The full message is "error from server. status code: 401 data from server: lifetime validation failed, the token is expired"